bndtools.m2e/src/bndtools/m2e/MavenFile.java

```diff
@@ -0,0 +1,299 @@
+package bndtools.m2e;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.nio.file.Path;
+
+@SuppressWarnings("serial")
+class MavenFile extends File {
```

Comment:
I would not extend File. That is super risky. I am sure it is not meant to be subclassed.

Reply:
fixed